src/components/pages/History/PoolShares.tsx

```diff
@@ -1,86 +1,188 @@
-import { useMetadata, useOcean } from '@oceanprotocol/react'
+import { useOcean } from '@oceanprotocol/react'
 import React, { ReactElement, useEffect, useState } from 'react'
 import Table from '../../atoms/Table'
-import { DDO, Logger, MetadataCache } from '@oceanprotocol/lib'
-import PriceUnit from '../../atoms/Price/PriceUnit'
 import Conversion from '../../atoms/Price/Conversion'
 import styles from './PoolShares.module.css'
 import AssetTitle from '../../molecules/AssetListTitle'
+import { gql, useQuery } from '@apollo/client'
+import {
+  PoolShares as PoolSharesList,
+  PoolShares_poolShares as PoolShare,
+  PoolShares_poolShares_poolId_tokens as PoolSharePoolIdTokens
+} from '../../../@types/apollo/PoolShares'
+import web3 from 'web3'
+import Token from '../../organisms/AssetActions/Pool/Token'
+
+const poolSharesQuery = gql`
+  query PoolShares($user: String) {
+    poolShares(where: { userAddress: $user, balance_gt: 0.001 }, first: 1000) {
+      id
+      balance
+      userAddress {
+        id
+      }
+      poolId {
+        id
+        datatokenAddress
+        lockedValue
+        tokens {
+          tokenId {
+            symbol
+          }
+        }
+        oceanReserve
+        datatokenReserve
+        totalShares
+        consumePrice
+        spotPrice
+      }
+    }
+  }
+`
 
 interface Asset {
-  ddo: DDO
-  shares: string
+  userLiquidity: number
+  poolShare: PoolShare
+}
+
+function calculateUserLiquidity(poolShare: PoolShare) {
+  const ocean =
+    (poolShare.balance / poolShare.poolId.totalShares) *
+    poolShare.poolId.oceanReserve
+  const datatokens =
+    (poolShare.balance / poolShare.poolId.totalShares) *
+    poolShare.poolId.datatokenReserve
+  const totalLiquidity = ocean + datatokens * poolShare.poolId.consumePrice
+  return totalLiquidity
+}
+
+function findValidToken(tokens: PoolSharePoolIdTokens[]) {
+  const symbol = tokens.find((token) => token.tokenId !== null)
+  return symbol.tokenId.symbol
 }
 
-function TotalLiquidity({ ddo }: { ddo: DDO }): ReactElement {
-  const { price } = useMetadata(ddo)
-  const totalLiquidityInOcean = price?.ocean + price?.datatoken * price?.value
+function Symbol({ tokens }: { tokens: PoolSharePoolIdTokens[] }) {
+  return <>{findValidToken(tokens)}</>
+}
 
+function YourLiquidity({ row }: { row: Asset }) {
```

Comment:
just a small architectural thing we don't necessarily need to bother with right now to not add more scope but would remove duplication and is more of a React-y-way: `YourLiquidity` and `PoolLiquidity` components should be just one component e.g. `Liquidity` with a `type="user" and `type="pool"` prop. Based on that inside of components we switch values based on that. Cause both have the exact same markup, and only the value change